controllers/apps/operations/expose.go

```diff
@@ -270,11 +270,16 @@ func (e ExposeOpsHandler) buildClusterServices(reqCtx intctrlutil.RequestCtx,
 			return nil, fmt.Errorf("component service is not defined, expose operation is not supported, cluster: %s, component: %s", cluster.Name, clusterCompSpec.Name)
 		}
 		defaultServicePorts := make([]corev1.ServicePort, 0, len(compServices))
+		portsSet := make(map[int32]struct{}) // to avoid duplicate ports
```

Comment:
The key of the map should include both the protocol and the port, for example, ${protocol}-${port}.

Reply:
![image](https://github.com/apecloud/kubeblocks/assets/1765402/545d096c-1cd5-45b6-9187-f633938f70d4)